lib/core/src/display/world.rs

```diff
@@ -28,221 +42,262 @@ use eval_tt::*;
+        self.rc.borrow_mut().event_loop.add_callback(func)
+    }
+
+    pub fn mod_stats<F:FnOnce(&Stats)>(&self, f:F) {
+        f(&self.rc.borrow().stats);
+    }
+}
+
+impl<T> Add<T> for World where WorldData: Add<T> {
+    type Result = AddResult<WorldData,T>;
+    fn add(&mut self, t:T) -> Self::Result {
+        self.rc.borrow_mut().add(t)
+    }
+}
+
+impl Deref for World {
+    type Target = Rc<RefCell<WorldData>>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.rc
+    }
+}
+
+
+
+// ====================
+// === StatsMonitor ===
+// ====================
+
+#[derive(Clone,Debug)]
+pub struct StatsMonitor {
+    rc: Rc<RefCell<StatsMonitorData>>
+}
+
+impl StatsMonitor {
+    pub fn new(stats:&Stats) -> Self {
+        let rc = Rc::new(RefCell::new(StatsMonitorData::new(stats)));
+        Self {rc}
+    }
+
+    pub fn begin(&self) {
+        self.rc.borrow_mut().begin()
+    }
+
+    pub fn end(&self) {
+        self.rc.borrow_mut().end()
+    }
+}
+
+
+#[derive(Debug)]
+pub struct StatsMonitorData {
+    stats   : Stats,
+    monitor : Monitor,
+    panels  : Vec<Panel>
+}
+
+impl StatsMonitorData {
+    fn new(stats:&Stats) -> Self {
+        let stats       = stats.clone_ref();
+        let mut monitor = Monitor::new();
+        let panels = vec![
+            monitor.add( monitor::FrameTime          :: new()       ),
+            monitor.add( monitor::Fps                :: new()       ),
+            monitor.add( monitor::WasmMemory         :: new()       ),
+            monitor.add( monitor::GpuMemoryUsage     :: new(&stats) ),
+            monitor.add( monitor::DrawCallCount      :: new(&stats) ),
+            monitor.add( monitor::DataUploadCount    :: new(&stats) ),
+            monitor.add( monitor::DataUploadSize     :: new(&stats) ),
+            monitor.add( monitor::BufferCount        :: new(&stats) ),
+            monitor.add( monitor::SymbolCount        :: new(&stats) ),
+            monitor.add( monitor::ShaderCount        :: new(&stats) ),
+            monitor.add( monitor::ShaderCompileCount :: new(&stats) ),
+            monitor.add( monitor::SpriteSystemCount  :: new(&stats) ),
+            monitor.add( monitor::SpriteCount        :: new(&stats) ),
+        ];
+        Self {stats,monitor,panels}
+    }
+
+    fn begin(&mut self) {
+        for panel in &self.panels {
+            panel.begin();
+        }
+    }
+
+    fn end(&mut self) {
+        for panel in &self.panels {
+            panel.end();
+        }
+        self.monitor.draw();
+        self.stats.reset();
+    }
+}
+
+
+
+// =================
+// === WorldData ===
+// =================
+
+// === Definition ===
 
 /// World is the top-level structure managing several instances of `Workspace`.
 /// It is responsible for updating the system on every animation frame.
 #[derive(Derivative)]
 #[derivative(Debug(bound=""))]
-pub struct World {
-    pub workspaces      : OptVec<Workspace>,
+pub struct WorldData {
+    pub workspace       : Workspace,
```

Comment:
Fix docs, as the World no longer manage several workspaces.

And additionally, I don't see what is the difference in responsibilities of Workspace and World.

Reply:
`And additionally, I don't see what is the difference in responsibilities of Workspace and World.` - Very good observation. Basically I dont know yet how it would evolve yet. I would like to keep them separate now because I feel that we will get back to managing several workspaces sooner than latter, but this involves some major arch refactoring. Docs fixed.